tests/unit/NoticeOfWithdrawal.spec.ts

```diff
@@ -0,0 +1,148 @@
+import Vue from 'vue'
+import Vuetify from 'vuetify'
+import { createLocalVue, shallowMount } from '@vue/test-utils'
+import { createPinia, setActivePinia } from 'pinia'
+import { useRootStore } from '@/stores'
+import NoticeOfWithdrawal from '@/views/NoticeOfWithdrawal.vue'
+import { ConfirmDialog, PaymentErrorDialog, ResumeErrorDialog, SaveErrorDialog }
+  from '@/components/dialogs'
+import { Certify } from '@/components/common'
+import { DocumentDelivery } from '@bcrs-shared-components/document-delivery'
+import mockRouter from './mockRouter'
+import VueRouter from 'vue-router'
+import PlanOfArrangement from '@/components/NoticeOfWithdrawal/PlanOfArrangement.vue'
+import RecordToBeWithdrawn from '@/components/NoticeOfWithdrawal/RecordToBeWithdrawn.vue'
+import ReferenceNumber from '@/components/NoticeOfWithdrawal/ReferenceNumber.vue'
+import StaffPayment from '@/components/NoticeOfWithdrawal/StaffPayment.vue'
+import StaffRoleErrorDialog from '@/components/NoticeOfWithdrawal/StaffRoleErrorDialog.vue'
+
+// suppress various warnings:
+// - "Unknown custom element <affix>" warnings
+// - "$listeners is readonly"
+// - "Avoid mutating a prop directly"
+// ref: https://github.com/vuejs/vue-test-utils/issues/532
+Vue.config.silent = true
+
+Vue.use(Vuetify)
+const localVue = createLocalVue()
+localVue.use(VueRouter)
+setActivePinia(createPinia())
+const rootStore = useRootStore()
+
+// Prevent the warning "[Vuetify] Unable to locate target [data-app]"
+document.body.setAttribute('data-app', 'true')
+
+describe('Notice of Withdrawal view', () => {
+  beforeEach(() => {
+    // init store
+    rootStore.keycloakRoles = ['staff']
+  })
+
+  it('mounts the sub-components properly', async () => {
+    const $route = { query: { filingId: '0' }, params: { filingToBeWithdrawn: '12345' } }
+
```

Comment:
Could you please help: How to mock the params in route?
If I comment out Line 403 in NoticeOfWithdrawal.vue
(    filingToBeWithdrawn = this.$route.params.filingToBeWithdrawn || '')
the Unit tests run successfully.
With the route.params, it shows **TypeError: Cannot read properties of undefined (reading 'params')**
I tried some solutions but didn't figure out

Reply:
Would this work?

https://github.com/bcgov/business-filings-ui/blob/f9747e4b20d50da231dcd8a45d1feb45ffa5a498/tests/unit/AgmLocationChg.spec.ts#L117

Reply:
Maybe this?

```
const $route = { query: { filingId: '0' filingToBeWithdrawn: '12345' } }
```

Reply:
Still showing TypeError...Seems it can mock query but didn't work for param

Reply:
I wonder if the parameter format you used in routes.ts is supposed to be used as a prop?

See https://v3.router.vuejs.org/guide/essentials/passing-props.html

Reply:
Maybe this isn't the right way to do it:

![image](https://github.com/user-attachments/assets/7f68d75b-8563-4420-896b-8560e382544f)

Sample router call:

![image](https://github.com/user-attachments/assets/29c15b1a-9620-4792-864c-897ae745e97c)

https://v3.router.vuejs.org/guide/essentials/navigation.html#programmatic-navigation

Reply:
Or maybe our mock router (see mockRouter.ts) doesn't support your param (which might be fixable with another router option?)?

Sorry, looks like you're doing something new here and I don't know how to help.

Reply:
Arwen, if this is fixed, please resolve this conversation.